eras/conway/impl/testlib/Test/Cardano/Ledger/Conway/Binary/Golden.hs

```diff
@@ -101,3 +72,103 @@ goldenListRedeemers =
+              (DeserialiseFailure 4 "Empty list found, expected non-empty")
+          it "redeemers" . expectFailureOnEmptyField 5 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 4 "Empty list found, expected non-empty")
+          it "plutusV2Script" . expectFailureOnEmptyField 6 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 4 "Empty list of scripts is not allowed")
+          it "plutusV3Script" . expectFailureOnEmptyField 7 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 4 "Empty list of scripts is not allowed")
+          it "8th field" . expectFailureOnEmptyField 8 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 2 "An error occured while decoding (Int,Void) not a valid key:.\nError: 8")
+        describe "Tagged" $ do
+          let
+            expectFailureOnEmptyField k =
+              expectDecoderFailureAnn @(TxWits era) version (witsEmptyFieldWithTag k)
+          it "addrTxWits" . expectFailureOnEmptyField 0 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list found, expected non-empty")
+          it "nativeScripts" . expectFailureOnEmptyField 1 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list found, expected non-empty")
+          it "bootstrapWitness" . expectFailureOnEmptyField 2 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list found, expected non-empty")
+          it "plutusV1Script" . expectFailureOnEmptyField 3 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list of scripts is not allowed")
+          it "plutusData" . expectFailureOnEmptyField 4 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list found, expected non-empty")
+          it "plutusV2Script" . expectFailureOnEmptyField 6 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list of scripts is not allowed")
+          it "plutusV3Script" . expectFailureOnEmptyField 7 $
+            DecoderErrorDeserialiseFailure
+              (Binary.label $ Proxy @(Annotator (TxWits era)))
+              (DeserialiseFailure 7 "Empty list of scripts is not allowed")
```

Comment:
"If you find yourself copy pasting large chunks of code, it's likely you doing something wrong" - this was a priceless advice I learned from a professor at the university.
This whole thing is unnecessarily duplicated:
```suggestion
```